### PR TITLE
encrypt bottlerocket second volume

### PR DESCRIPTION
**Pull request by bwagner5**
**1. Issue, if available:**
https://github.com/aws/karpenter/issues/1408

**2. Description of changes:**
 - Bottlerocket mounts two volumes, xvda is for bottlerocket system components (like the API, OS, bootloader, etc) and xvdb is for containers (container images, logs, etc). 

**3. How was this change tested?**
 - Launched Bottlerocket node before the change to reproduce the second volume being unencrypted
 - Launched Bottlerocket node w/ change and observed both volumes are now encrypted 

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/cloudprovider/aws/launchtemplate.go [hunk 1]
```diff
@@ -182,17 +182,39 @@ func needsDocker(is []cloudprovider.InstanceType) bool {
 	return false
 }
 
-func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, options *launchTemplateOptions) (*ec2.LaunchTemplate, error) {
-	output, err := p.ec2api.CreateLaunchTemplateWithContext(ctx, &ec2.CreateLaunchTemplateInput{
-		LaunchTemplateName: aws.String(launchTemplateName(options)),
-		LaunchTemplateData: &ec2.RequestLaunchTemplateData{
-			BlockDeviceMappings: []*ec2.LaunchTemplateBlockDeviceMappingRequest{{
+func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, constraints *v1alpha1.Constraints, options *launchTemplateOptions) (*ec2.LaunchTemplate, error) {
+	blockDeviceMappings := []*ec2.LaunchTemplateBlockDeviceMappingRequest{{
```
Comment (ellistarn) [inline pkg/cloudprovider/aws/launchtemplate.go:186]: Maybe worth factoring out into a method, which will make things easier when we expose this to users.

**Review by ellistarn — APPROVED** [review]
Nice fix! Needs docs?

File: pkg/cloudprovider/aws/launchtemplate.go (hunk 1 above)
Comment (bwagner5) [reply pkg/cloudprovider/aws/launchtemplate.go:186]: planning on refactoring when we support the provisioner root volume configuration

**Comment by bwagner5** [comment]
> Nice fix! Needs docs?

We probably do need docs on the general feature of Karpenter encrypting volumes by default. I'll put in an issue for those docs. I guess they'd live in AWS/Provisioning.

**Pull request merged**